llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -11841,9 +11841,10 @@ void BoUpSLP::reorderInputsAccordingToOpcode(ArrayRef<Value *> VL,
 }
 
 Instruction &BoUpSLP::getLastInstructionInBundle(const TreeEntry *E) {
-  auto &Res = EntryToLastInstruction.FindAndConstruct(E);
-  if (Res.second)
-    return *Res.second;
+  auto It = EntryToLastInstruction.find(E);
+  if (It != EntryToLastInstruction.end())
+    return *It->second;
+  auto &Res = EntryToLastInstruction.try_emplace(E).first->second;
```

Comment:
Why does this use find and then try_emplace? This will perform two hash lookups. It's enough to use only try_emplace and change behavior depending on the inserted boolean.

Reply:
Thank you for catching this!  I'm addressing this with https://github.com/llvm/llvm-project/pull/107143.